docs/ADD_NEW_EXCHANGE.md

```diff
@@ -0,0 +1,1064 @@
+
+https://docs.ftx.com/#fills-2
+
+```go
+// WsFills stores websocket fills' data
+type WsFills struct {
+	Fee       float64   `json:"fee"`
+	FeeRate   float64   `json:"feeRate"`
+	Future    string    `json:"future"`
+	ID        int64     `json:"id"`
+	Liquidity string    `json:"liquidity"`
+	Market    string    `json:"market"`
+	OrderID   int64     `json:"int64"`
+	TradeID   int64     `json:"tradeID"`
+	Price     float64   `json:"price"`
+	Side      string    `json:"side"`
+	Size      float64   `json:"size"`
+	Time      time.Time `json:"time"`
+	OrderType string    `json:"orderType"`
+}
+
+// WsFillsDataStore stores ws fills' data
+type WsFillsDataStore struct {
+	Channel     string  `json:"channel"`
+	MessageType string  `json:"type"`
+	FillsData   WsFills `json:"fills"`
+}
+```
+
+- Create the authentication function based on specifications provided in the documentation:
+
+https://docs.ftx.com/#private-channels
+
+```go
+// WsAuth sends an authentication message to receive auth data
+func (f *FTX) WsAuth() error {
+	intNonce := time.Now().UnixNano() / 1000000
+	strNonce := strconv.FormatInt(intNonce, 10)
+	hmac := crypto.GetHMAC(
+		crypto.HashSHA256,
+		[]byte(strNonce+"websocket_login"),
+		[]byte(f.API.Credentials.Secret),
+	)
+	sign := crypto.HexEncodeToString(hmac)
+	req := Authenticate{Operation: "login",
+		Args: AuthenticationData{
+			Key:  f.API.Credentials.Key,
+			Sign: sign,
+			Time: intNonce,
+		},
+	}
+	return f.WebsocketConn.SendJSONMessage(req)
+}
+```
+
+- Create an unsubscribe function if the exchange has the functionality:
+
+```go
+// Unsubscribe sends a websocket message to stop receiving data from the channel
+func (f *FTX) Unsubscribe(channelToSubscribe wshandler.WebsocketChannelSubscription) error {
+	var unSub WsSub
+	a, err := f.GetPairAssetType(channelToSubscribe.Currency)
+	if err != nil {
+		return err
+	}
+	unSub.Operation = "unsubscribe"
+	unSub.Channel = channelToSubscribe.Channel
+	unSub.Market = f.FormatExchangeCurrency(channelToSubscribe.Currency, a).String()
+	return f.WebsocketConn.SendJSONMessage(unSub)
+}
+```
+
+#### Complete websocket setup in wrapper:
+
+```go
+	err = f.Websocket.Setup(
+		&wshandler.WebsocketSetup{
+			Enabled:                          exch.Features.Enabled.Websocket,
+			Verbose:                          exch.Verbose,
+			AuthenticatedWebsocketAPISupport: exch.API.AuthenticatedWebsocketSupport,
+			WebsocketTimeout:                 exch.WebsocketTrafficTimeout,
+			DefaultURL:                       ftxWSURL,
+			ExchangeName:                     exch.Name,
+			RunningURL:                       exch.API.Endpoints.WebsocketURL,
+			Connector:                        f.WsConnect,
+			Subscriber:                       f.Subscribe,
+			UnSubscriber:                     f.Unsubscribe,
+			Features:                         &f.Features.Supports.WebsocketCapabilities,
+		})
+	if err != nil {
+		return err
+  }
+  ```
+
+  ```go
```

Comment:
Please add a description of what the the below entails (matching up the features supported by each protocol and whether the exchange supports websocket/REST etc